Sources/Hyperspace/Service/Network/Cancellation.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Cancellation.swift
+//  Hyperspace
+//
+//  Created by Will McGinty on 2/19/19.
+//  Copyright © 2019 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+/// Manages cancellation tokens and signals them when cancellation is requested.
+///
+/// All `CancellationTokenSource` methods are thread safe.
+public final class CancellationSource {
+    
+    public struct Token {
+        // MARK: Properties
+        private weak var source: CancellationSource?
```

Comment:
Seems strange to me that the token needs a reference to the source.

Reply:
This is so the closure used in the `register` method can be passed back to the `CancellationSource` so that synchronized cancellation can occur.